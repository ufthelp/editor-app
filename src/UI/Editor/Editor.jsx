import React, {Component} from "react";
import AceEditor from "react-ace";
import CustomLanguageMode from '../../util/Groovy/GroovyLanguageMode';
//import CustomLanguageMode from '../../util/SQL/SQLLanguageMode';

import 'brace/mode/sql'; //for rendering language = java in editor
import 'brace/mode/groovy';
import 'brace/theme/twilight'; //theme variable
import 'brace/ext/language_tools' //autocompletion

import "./Editor.scss";

class Editor extends Component {
  aceEditor= React.createRef(); //ref to editor element
  
  componentDidMount() {
    const customMode = new CustomLanguageMode(); //custom rules, syntax highlighting
    this.aceEditor.current.editor.getSession().setMode(customMode)

    //autocompletion logic
    let langTools = window.ace.acequire('ace/ext/language_tools')
    langTools.setCompleters([langTools.snippetCompleter, langTools.keyWordCompleter])
  }
 
  render(){
    const {...props} = this.props;
    return(
        <AceEditor {...props}  
            ref={this.aceEditor}
         />
    )
  }
}


export default Editor;
