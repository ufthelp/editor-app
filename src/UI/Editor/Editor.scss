.ace-multi-cursor {
    pointer-events: auto;
}