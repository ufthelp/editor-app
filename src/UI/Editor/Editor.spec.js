import React from 'react';
import {shallow} from 'enzyme';
import Editor from './Editor';


describe('<Editor /> Component', ()=>{
    let wrapper;
    beforeEach(()=> wrapper = shallow( <Source 
        fontSize="14"
    />))

    it('should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('should render the value of title prop', () => {
        wrapper.setProps({ value: 'file' });
        expect(wrapper.text()).toEqual('<ReactAce />');
    });
})

