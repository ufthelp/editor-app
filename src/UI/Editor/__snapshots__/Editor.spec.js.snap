// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Editor /> Component should render correctly 1`] = `undefined`;
